projects/core/directives/icons/icons.directive.ts

```diff
@@ -35,7 +42,7 @@ export class TuiIcons {
     protected readonly resolver: TuiStringHandler<string> = tuiInjectIconResolver();
 
     @Input()
-    public iconStart = inject(TUI_ICON_START, {self: true, optional: true}) || '';
+    public iconStart: Icon = inject(TUI_ICON_START, {self: true, optional: true}) || '';
 
     @Input()
     public iconEnd = inject(TUI_ICON_END, {self: true, optional: true}) || '';
```

Comment:
shouldn't be as consistent?

`public iconEnd: Icon`

Reply:
No, no reason to add hack here